consumer/batching.go

```diff
@@ -3,77 +3,276 @@ package consumer
 import (
 	"fmt"
 	"github.com/deso-protocol/core/lib"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"math"
+	"time"
 )
 
-type BatchedEntry struct {
-	Encoder *lib.DeSoEncoder
-	Key     []byte
+const (
+	retryLimit = 10
+)
+
+// BatchIndex is a struct that contains the index of the first entry in a batch, and the batch's index relative to all other executed batches.
+type BatchIndexInfo struct {
```

Comment:
Elaborate on this comment.

Reply:
It would be nice to have a big fat explanation at the top of all the minutae of how the batches are made and stuff.